src/main/scala/com/microsoft/hyperspace/index/CachingIndexCollectionManager.scala

```diff
@@ -59,11 +60,16 @@ class CachingIndexCollectionManager(
    * @return list of indexes whose current state is among desired states
    */
   override def getIndexes(states: Seq[String] = Seq()): Seq[IndexLogEntry] = {
-    indexCache.get().getOrElse {
-      val originalIndexes = super.getIndexes(states)
-      indexCache.set(originalIndexes)
-      originalIndexes
-    }
```

Comment:
It might be better to cache indexes with all states and return filtered indexes for the given states

Reply:
I considered that too but didn't choose it because it relies on the semantics of getIndexes. On the other hand, this version works regardless of how getIndexes works.

However, it seems we have to do as you said due to caching (tagging) in IndexLogEntry.